R/prune.R

```diff
@@ -0,0 +1,158 @@
+##' @importFrom utils capture.output
+
+extract_prune <- function(e, path, con) {
+
+  # The prune.csv file indicates...
+  # modelling_group, disease, scenario, and touchstones
+
+  # these can be '*' for wildcard,
+  # or a semi-colon separated list of options.
+
+  # No CSV provided
+  if (is.null(e$prune_csv)) {
+    return(NULL)
+  }
+
+  # Empty CSV provided
+  if (nrow(e$prune_csv) == 0) {
+    return(NULL)
+  }
+
+  # Check all necessary things exist, and
+  # return vector of things.
+
+  thing_exists <- function(entries, table) {
+    entries <- unique(unlist(strsplit(entries, ";")))
+    if ("*" %in% entries) {
+      return(NULL)
+    }
+    db_entries <- DBI::dbGetQuery(con, sprintf("
+      SELECT * FROM %s WHERE id IN ('%s')", table,
+        paste(entries, collapse="','")))$id
+    if (any(!entries %in% db_entries)) {
+      entries <- paste(entries[!entries %in% db_entries],
+                       collapse = ", ")
+      stop(sprintf("%s not found: %s\n", table, entries))
+    }
+    db_entries
+  }
+
+  modelling_group <- thing_exists(e$prune_csv$modelling_group, "modelling_group")
+  disease <- thing_exists(e$prune_csv$disease, "disease")
+  touchstone <- thing_exists(e$prune_csv$scenario, "scenario_description")
+  scenario <- thing_exists(e$prune_csv$touchstone, "touchstone")
+
+
+  # For each row in the CSV file:
+
+  burden_estimate_set <- NULL
+
+  for (i in seq_len(nrow(e$prune_csv))) {
+    row <- e$prune_csv[i, ]
+    touchstone <- unlist(strsplit(row$touchstone, ";"))
+    modelling_group <- unlist(strsplit(row$modelling_group, ";"))
+    disease < unlist(strsplit(row$disease, ";"))
+    scenario <- unlist(strsplit(row$scenario, ";"))
```

Comment:
Could put this in a little helper function as it is used here and in `thing_exists`

Reply:
Ah yes - and it seems I already have "split_semi" for this